bindata/etcd/cluster-restore.sh

```diff
@@ -53,6 +53,32 @@ function restore_static_pods() {
     tar -xvf "${BACKUP_FILE}" --strip-components=2 -C "${MANIFEST_DIR}"/ "${BACKUP_POD_PATH}"
   done
 }
+function replace_node_specific_parameters() {
+  pod_file="$1"
+# shellcheck disable=SC2016
+  declare -A SUBS_MAP
+  SUBS_MAP['export ETCD_NAME=.*$']='export ETCD_NAME=${NODE_NODE_ENVVAR_NAME_ETCD_NAME}'
```

Comment:
Maybe what you're proposing here is considerably more complicated than just figuring out multi-line sed?

Reply:
Well, this is just an associative array. Sam wanted to make sure the substitution patterns exist before we attempt to replace them. So, this achieves that.

Reply:
I defer to @marun

/assign @marun

Reply:
@retroflexer If I recall our conversation correctly from last week, 

 - discover what will be replaced
 - multiline sed
 - I suggest checking that the substitution patterns exists before replacement with `grep -q <substitution pattern>`. Note that with